at_client/src/main/java/org/atsign/client/util/RegisterUtil.java

```diff
@@ -0,0 +1,108 @@
+            dataJsonObject = dataJsonObject.getJSONObject("data");
+            return dataJsonObject.getString("atsign");
+        } else {
+            throw new AtException(connection.getResponseCode() + " " + connection.getResponseMessage());
+        }
+    }
+
+    //Accepts your email and an unpaired atsign. This method will pair the free atsign with your email.
+    //Sends the one-time-password to the provided email. Returns bool, true if OTP sent or False otherwise.
+    public Boolean registerAtsign(String email, String atsign) throws AtException, MalformedURLException, IOException {
+        URL urlObject = new URL(Constants.AT_DEV_DOMAIN + Constants.API_PATH + Constants.REGISTER_ATSIGN);
+        HttpsURLConnection httpsConnection = (HttpsURLConnection) urlObject.openConnection();
+        String params = "{\"atsign\":\"" + atsign + "\", \"email\":\"" + email + "\"}";
+        httpsConnection.setRequestMethod("POST");
+        httpsConnection.setRequestProperty("Content-Type", "application/json");
+        httpsConnection.setRequestProperty("Authorization", Constants.DEV_API_KEY);
+        httpsConnection.setDoOutput(true);
+        OutputStream outputStream = httpsConnection.getOutputStream();
+        outputStream.write(params.getBytes(StandardCharsets.UTF_8), 0, params.length());
+        outputStream.flush();
+        outputStream.close();
+        System.out.println(httpsConnection.getResponseCode());
+        System.out.println(httpsConnection.getResponseMessage());
+        if (httpsConnection.getResponseCode() == HttpsURLConnection.HTTP_OK) {
+            BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(
+                    httpsConnection.getInputStream()));
+            StringBuffer response = new StringBuffer();
+            response.append(bufferedReader.readLine());
+            if (response.toString().contains("Sent Successfully")) {
+                return true;
+            }
+            return false;
+        }
+        throw new AtException(httpsConnection.getResponseCode() + " " + httpsConnection.getResponseMessage());
+    }
+
+    //Accepts your email, unpaired atsign, and the otp received on the provided email.
+    //Validates the one-time-password against the atsign and registers it to the provided email if valid.
+    //Returns the CRAM secret pertaining to the atsign which is registered.
+    public String validateOtp(String email, String atsign, String otp) throws IOException, AtException {
+        String validationUrl = Constants.AT_DEV_DOMAIN + Constants.API_PATH + Constants.VALIDATE_OTP;
+        URL validateOtpUrl = new URL(validationUrl);
+        HttpsURLConnection httpsConnection = (HttpsURLConnection) validateOtpUrl.openConnection();
+        String params = "{\"atsign\":\"" + atsign + "\", \"email\":\"" + email + "\", \"otp\":\"" + otp
+                + "\", \"confirmation\":\"" + "true\"}";
+        System.out.println(params);
```

Comment:
These Sysouts on a CLI might be a hinderance. Pls check the rest of the Gary's code on usage of print statements.